### PR TITLE
jcf_collections/ it-academy.by/ JC1-110-22 /Anastasia Melnikova/ Done project StringCollectionUtil and List

### PR DESCRIPTION
**Pull request by Stellmaria**
## Pull request state

Actual status for PR:
- [x] implemented
- [x] reviewed
- [x] fixed
- [ ] approved

### CODE REVIEW

File: src/main/java/com/rakovets/course/java/core/practice/jcf_list/clazz/Person.java [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+
+    @Override
+    public String toString() {
+        return "Person{" +
+                "name='" + name + '\'' +
+                ", surname='" + surname + '\'' +
+                ", age=" + age +
+                ", averageAnnualMark=" + averageAnnualMark +
+                '}';
+    }
```
Comment (rakovets) [inline src/main/java/com/rakovets/course/java/core/practice/jcf_list/clazz/Person.java:52]: move to bottom

File: src/main/java/com/rakovets/course/java/core/practice/jcf_list/clazz/SchoolClass.java [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+    public Person getBestStudent(List<Person> students) {
+        students.sort(new AverageAnnualMarkComparator());
+        return students.get(students.size() - 1);
+    }
+
+    /**
+     * Comparator by name.
+     */
+    public static class NameComparator implements Comparator<Person> {
+
```
Comment (rakovets) [inline src/main/java/com/rakovets/course/java/core/practice/jcf_list/clazz/SchoolClass.java:34]: remove it

File: src/main/java/com/rakovets/course/java/core/practice/jcf_list/clazz/SchoolClass.java [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+        public int compare(Person o1, Person o2) {
+            return o1.getFullName().compareTo(o2.getFullName());
+        }
+    }
+
+    /**
+     * Comparator by age.
+     */
+    public static class AgeComparator implements Comparator<Person> {
+
```
Comment (rakovets) [inline src/main/java/com/rakovets/course/java/core/practice/jcf_list/clazz/SchoolClass.java:45]: remove it

File: src/main/java/com/rakovets/course/java/core/practice/jcf_list/clazz/SchoolClass.java [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+        public int compare(Person o1, Person o2) {
+            return Integer.compare(o1.getAge(), o2.getAge());
+        }
+    }
+
+    /**
+     * Comparator by average annual mark.
+     */
+    public static class AverageAnnualMarkComparator implements Comparator<Person> {
+
```
Comment (rakovets) [inline src/main/java/com/rakovets/course/java/core/practice/jcf_list/clazz/SchoolClass.java:56]: remove it

File: src/test/java/com/rakovets/course/java/core/practice/jcf_collection/StringCollectionUtilTest.java [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+
+    @Test
+    @DisplayName("Remove words by length.")
+    void removeWordsByLength() {
+        String[] str2 = new String[]{"this", "lots", "fun", "for", "every", "Java", "programmer"};
+
+        StringCollectionUtil.removeWordsByLength(stringCollection, 2);
+
+        Assertions.assertEquals(new ArrayList<>(Arrays.asList(str2)), stringCollection);
+    }
```
Comment (rakovets) [inline src/test/java/com/rakovets/course/java/core/practice/jcf_collection/StringCollectionUtilTest.java:39]: fix it

File: src/test/java/com/rakovets/course/java/core/practice/jcf_list/StudioTest.java [hunk 6]
```diff
@@ -0,0 +1,126 @@
...
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+@DisplayName("Test for class Studio.")
+class StudioTest {
+
```
Comment (rakovets) [inline src/test/java/com/rakovets/course/java/core/practice/jcf_list/StudioTest.java:16]: remove it

**Review by rakovets — CHANGES_REQUESTED** [review]
need to fix

File: src/main/java/com/rakovets/course/java/core/practice/jcf_collection/StringCollectionUtil.java [hunk 7]
```diff
@@ -0,0 +1,36 @@
...
+
+interface StringCollectionUtil {
+    /**
+     * The method replaces the original string, the length of which is equal to the size of the word, with *.
+     *
+     * @param collection string list.
+     * @param size       word size.
+     * @return list after changes.
+     */
+    static Collection<String> resetWordsByLength(Collection<String> collection, int size) {
```
[the remark this answers is not in the document]
Comment (Stellmaria) [reply src/main/java/com/rakovets/course/java/core/practice/jcf_collection/StringCollectionUtil.java]: fixed

[the remark this answers is not in the document]
Comment (Stellmaria) [reply src/main/java/com/rakovets/course/java/core/practice/jcf_collection/StringCollectionUtil.java]: fixed

File: src/main/java/com/rakovets/course/java/core/practice/jcf_collection/StringCollectionUtil.java [hunk 8]
```diff
@@ -0,0 +1,36 @@
...
+        return stringCollection;
+    }
+
+    /**
+     * The method removes from the list of strings all strings with a length equal to the dimension of the word.
+     *
+     * @param collection string list.
+     * @param size       word size.
+     */
+    static void removeWordsByLength(Collection<String> collection, int size) {
```
[the remark this answers is not in the document]
Comment (Stellmaria) [reply src/main/java/com/rakovets/course/java/core/practice/jcf_collection/StringCollectionUtil.java]: fixed

File: src/main/java/com/rakovets/course/java/core/practice/jcf_list/clazz/Person.java [hunk 9]
```diff
@@ -0,0 +1,73 @@
...
+    private final double averageAnnualMark;
+
+    /**
+     * Constructor.
+     *
+     * @param name              student's name.
+     * @param surname           last name of the student.
+     * @param age               student's age.
+     * @param averageAnnualMark average student score.
+     */
```
[the remark this answers is not in the document]
Comment (Stellmaria) [reply src/main/java/com/rakovets/course/java/core/practice/jcf_list/clazz/Person.java]: fixed

File: src/main/java/com/rakovets/course/java/core/practice/jcf_list/clazz/Person.java (hunk 1 above)
Comment (Stellmaria) [reply src/main/java/com/rakovets/course/java/core/practice/jcf_list/clazz/Person.java:64]: fixed

File: src/main/java/com/rakovets/course/java/core/practice/jcf_list/clazz/SchoolClass.java (hunk 2 above)
Comment (Stellmaria) [reply src/main/java/com/rakovets/course/java/core/practice/jcf_list/clazz/SchoolClass.java]: fixed

File: src/main/java/com/rakovets/course/java/core/practice/jcf_list/clazz/SchoolClass.java (hunk 3 above)
Comment (Stellmaria) [reply src/main/java/com/rakovets/course/java/core/practice/jcf_list/clazz/SchoolClass.java:23]: fixed

File: src/main/java/com/rakovets/course/java/core/practice/jcf_list/clazz/SchoolClass.java (hunk 4 above)
Comment (Stellmaria) [reply src/main/java/com/rakovets/course/java/core/practice/jcf_list/clazz/SchoolClass.java]: fixed

File: src/test/java/com/rakovets/course/java/core/practice/jcf_collection/StringCollectionUtilTest.java (hunk 5 above)
Comment (Stellmaria) [reply src/test/java/com/rakovets/course/java/core/practice/jcf_collection/StringCollectionUtilTest.java]: fixed

File: src/test/java/com/rakovets/course/java/core/practice/jcf_list/StudioTest.java (hunk 6 above)
Comment (Stellmaria) [reply src/test/java/com/rakovets/course/java/core/practice/jcf_list/StudioTest.java:17]: fixed

**Pull request closed without merging**